types/types.go

```diff
@@ -369,6 +369,7 @@ type (
 		// WithNullability returns a copy of this type but with
 		// the nullability set to the passed in value
 		WithNullability(Nullability) Type
+		Match(ot Type) bool
```

Comment:
how is Match different from Equals? It looks like most of the implementations make it identical?

If `Match` is different from `Equals` then we should add a comment explaining why you would use one over the other.

Reply:
I fixed PR to have "MatchWithNullability" and "MatchWithoutNullability" methods instead of "Match" method.
I missed it earlier. Based on Option Nullability on function, either MatchWithNullability or MatchWithoutNullability will be called.
So considering this, Match* and equal won't be same